frontend/src/pages/notebookController/screens/server/usePreferredNotebookSize.ts

```diff
@@ -1,29 +1,52 @@
 import * as React from 'react';
-import AppContext from '../../../../app/AppContext';
+import { useAppContext } from '../../../../app/AppContext';
 import { useNotebookUserState } from '../../../../utilities/notebookControllerUtils';
+import { DEFAULT_NOTEBOOK_SIZES } from '../../const';
+import { NotebookSize } from '../../../../types';
+import useNotification from '../../../../utilities/useNotification';
 
-export const usePreferredNotebookSize = (): [
-  selectedSize: string,
-  setSelectedSize: (size: string) => void,
-] => {
-  const { dashboardConfig } = React.useContext(AppContext);
+export const usePreferredNotebookSize = (): {
+  selectedSize: NotebookSize;
+  setSelectedSize: (size: string) => void;
+  sizes: NotebookSize[];
+} => {
+  const { dashboardConfig } = useAppContext();
+  const notification = useNotification();
+  const sizes = dashboardConfig?.spec?.notebookSizes || DEFAULT_NOTEBOOK_SIZES;
```

Comment:
An empty array is truthy -- you crash the UI with no notebooks -- lets call no notebookSizes the same as not being defined.